### PR TITLE
Dedicated executor docs

### PR DESCRIPTION
**Pull request by Krizzu**
Summary:
---------

Docs explaining the usage of a dedicated thread executor for Async Storage.

### CODE REVIEW

File: example/android/gradle.properties [hunk 1]
```diff
@@ -18,4 +18,6 @@
 # org.gradle.parallel=true
 
 # This is an example of how you can change default DB size (6MB) to 10MB
-# AsyncStorage_db_size_in_MB=10
```
Comment (thymikee) [inline example/android/gradle.properties]: deleted without the comment above

File: docs/advanced/CustomExecutor.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Custom Thread Executor
+
+Android only feature
+
+## Motivation
+
+By default, Async Storage is executed in a background thread, using Thread Executor provided by `AsyncTask` utility. Being a singleton, it puts a task that has to be executed in a queue, while waiting for a free thread (from its fixed-size pool) to use.
```
Comment (thymikee) [inline docs/advanced/CustomExecutor.md]:
```suggestion
By default, Async Storage is executed in a background thread, using Thread Executor provided by `AsyncTask` utility. Executor is a singleton queuing a task that has to be executed, while waiting for a free thread (from its fixed-size pool) to use. 
```

File: docs/advanced/CustomExecutor.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Custom Thread Executor
+
+Android only feature
+
+## Motivation
+
+By default, Async Storage is executed in a background thread, using Thread Executor provided by `AsyncTask` utility. Being a singleton, it puts a task that has to be executed in a queue, while waiting for a free thread (from its fixed-size pool) to use. 
+
+For apps (or libraries) that use said executor in more than one place, this increases the possibility of awaiting for a free thread.
```
Comment (thymikee) [inline docs/advanced/CustomExecutor.md]:
```suggestion
For apps or libraries that use said executor in more than one place, e.g. "brownfield" apps, such scenario increases the possibility of a thread starvation, slowing down the app. By turning on a "custom executor", AsyncStorage will be instantiated with a dedicated thread executor with its own thread pool, lowering the chance of starvation.
```

File: docs/advanced/CustomExecutor.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+Android only feature
+
+## Motivation
+
+By default, Async Storage is executed in a background thread, using Thread Executor provided by `AsyncTask` utility. Being a singleton, it puts a task that has to be executed in a queue, while waiting for a free thread (from its fixed-size pool) to use. 
+
+For apps (or libraries) that use said executor in more than one place, this increases the possibility of awaiting for a free thread. 
+
+If you noticed a delay in accessing your storage, this flag might help you - it allows you to use a dedicated executor for a Async Storage.
```
Comment (thymikee) [inline docs/advanced/CustomExecutor.md]: Remove
```suggestion
```

**Comment by Krizzu** [comment]
@thymikee 	thanks.
I changed the whole thing, so I discarded most of your comments. I modified the feature name + flag name, for a better resemblance.

**Pull request merged**